Modules/IO/NIFTI/include/itkNiftiImageIO.h

```diff
@@ -131,10 +156,10 @@ class ITKIONIFTI_EXPORT NiftiImageIO : public ImageIOBase
 
   /** A mode to allow the Nifti filter to read and write to the LegacyAnalyze75 format as interpreted by
    * the nifti library maintainers.  This format does not properly respect the file orientation fields.
-   * By default this is set to true.
+   * By default this is set by configuration option ITK_NIFTI_IO_ANALYZE_FLAVOR
    */
-  itkSetMacro(LegacyAnalyze75Mode, bool);
-  itkGetConstMacro(LegacyAnalyze75Mode, bool);
+  itkSetMacro(LegacyAnalyze75Mode, Analyze75Flavor);
```

Comment:
We are all ok with changing this public API's signature?

Reply:
I think this API was added since last release, so only people living on `master` will feel the effect of this change.

Reply:
In fact, no, that API is present in ITK 4.13.2, and probably earlier too.

Reply:
OK, thanks for clarification Sean. I am fine with changing the signature, as people should have moved away from Analyze files. But we should document the change in [migration guide](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Documentation/ITK5MigrationGuide.md).

Reply:
@vfonov and @seanm, can one of you make a PR with migration guide update?

Reply:
In fact, if you've "moved away from Analyze files" then you might already be deliberately calling SetLegacyAnalyze75Mode(false) to make sure you *don't* open Analyze files. Now that won't compile.

Reply:
Well, a PR which restores the old API, and gives the new API a new name is a possible alternative. As I never used Analyze files, I don't have much preference for one over the other. I believe most people just don't use IOs directly, and those who do, mostly use GDCMImageIO.

Reply:
It may be useful to break the API to make sure that developers notice the potential change in behavior. A comment can be added near where the build error occurs that points to corresponding section in the migration guide.